packages/core/src/components/widgets/SelectWidget.js

```diff
@@ -63,41 +61,40 @@ function SelectWidget(props) {
     placeholder,
   } = props;
   const { enumOptions, enumDisabled } = options;
-  const emptyValue = multiple ? [] : "";
+  const selected = multiple
+    ? getMultipleSelected(value, enumOptions)
+    : getOneSelected(value, enumOptions);
   return (
     <select
       id={id}
       multiple={multiple}
       className="form-control"
-      value={typeof value === "undefined" ? emptyValue : value}
+      value={selected}
       required={required}
       disabled={disabled || readonly}
       autoFocus={autofocus}
       onBlur={
         onBlur &&
         (event => {
-          const newValue = getValue(event, multiple);
-          onBlur(id, processValue(schema, newValue));
+          onBlur(id, getValue(event, enumOptions, multiple));
         })
       }
       onFocus={
         onFocus &&
         (event => {
-          const newValue = getValue(event, multiple);
-          onFocus(id, processValue(schema, newValue));
+          onFocus(id, getValue(event, enumOptions, multiple));
         })
       }
       onChange={event => {
-        const newValue = getValue(event, multiple);
-        onChange(processValue(schema, newValue));
+        onChange(getValue(event, enumOptions, multiple));
       }}>
       {!multiple && schema.default === undefined && (
         <option value="">{placeholder}</option>
       )}
       {enumOptions.map(({ value, label }, i) => {
         const disabled = enumDisabled && enumDisabled.indexOf(value) != -1;
         return (
-          <option key={i} value={value} disabled={disabled}>
```

Comment:
This is likely a breaking change because it changes what's rendered in the DOM (the `value`)... but it's fine to go into version 2, and worth it to allow for better selection of any kinds of enums.